cores/arduino/stm32/spi_com.c

```diff
@@ -371,6 +371,8 @@ void spi_init(spi_t *obj, uint32_t speed, spi_mode_e mode, uint8_t msb)
   }
 #endif
 
+  handle->State = HAL_SPI_STATE_RESET;
```

Comment:
Could you move this line in the  `  /* Fill default value */` (start at line number 222)